### ISSUE TITLE
Improve the description of display_login_attempts rule and respective remediation

### ISSUE THREAD

**Issue by marcusburghardt** [ComplianceAsCode/content#9031]
#### Description of problem:

The `display_login_attempts` is mentioning the following statement in its description:
```
And make sure that the silent option is not set for pam_lastlog module.
```
- https://github.com/ComplianceAsCode/content/blob/master/linux_os/guide/system/accounts/accounts-pam/display_login_attempts/rule.yml#L17

This is probably inherited from STIG Requirement:
- https://www.stigviewer.com/stig/red_hat_enterprise_linux_8/2021-12-03/finding/V-230381
- https://www.stigviewer.com/stig/red_hat_enterprise_linux_8/2021-12-03/finding/V-230382

As it is described the most probable interpretation is that `silent` option should be removed in all `pam_lastlog` lines.

However, the intention of the requirement is to make the user aware of last access using its account and consequently giving him more information to notice any suspected access.

This could be achieved without removing the `silent` option from all lines. There are cases where the `silent` option is desired and safe to be kept while honoring the requirement goal. Removing it in all lines impact in user experience by creating avoidable noise.

#### SCAP Security Guide Version:

master

#### Operating System Version:

N/A

#### Steps to Reproduce:

1. Apply the `display_login_attempts` remediation
2. Login in the system and not that the last successful login information is duplicated.

#### Actual Results:

Last successful login information is duplicated.

#### Expected Results:

Last successful login information is not duplicated.

#### Additional Information/Debugging Steps:

This is the usual content of `/etc/pam.d/postlogin` file before the remediation:
```
session     optional                   pam_umask.so silent
session     [success=1 default=ignore] pam_succeed_if.so service !~ gdm* service !~ su* quiet
session     [default=1]                pam_lastlog.so nowtmp showfailed
session     optional                   pam_lastlog.so silent noupdate showfailed
```
In this default configuration, this is the output after a failed login followed by a successful login:
```
$ ssh root@rhel8
Last failed login: Wed Aug  9 08:31:44 EDT 2023 from 192.168.122.1 on ssh:notty
There was 1 failed login attempt since the last successful login.
Last login: Wed Aug  9 08:31:00 2023 from 192.168.122.1
```

This would be the content of `/etc/pam.d/postlogin` file after the remediation:
```
session     required                  pam_lastlog.so showfailed
session     optional                   pam_umask.so silent
session     [success=1 default=ignore] pam_succeed_if.so service !~ gdm* service !~ su* quiet
session     [default=1]                pam_lastlog.so nowtmp showfailed
session     optional                   pam_lastlog.so noupdate showfailed
```
It means that failed logins will be shown in any case, but successful login will be shown three times, one for each `pam_lastlog.so` line. The inclusion of the first line is even unnecessary to achieve requirement. Also, removing the silent in this last line will only create noise by repeating again information to the user, as shown in the example:

```
$ ssh root@rhel8
Last login: Wed Aug  9 08:31:52 EDT 2023 from 192.168.122.1 on pts/0
Last failed login: Wed Aug  9 08:35:39 EDT 2023 from 192.168.122.1 on ssh:notty
There were 3 failed login attempts since the last successful login.
Last login: Wed Aug  9 08:35:44 EDT 2023 from 192.168.122.1 on ssh
Last login: Wed Aug  9 08:35:44 2023 from 192.168.122.1
```

By not removing the `silent` option on the last line, this would be the content of `/etc/pam.d/postlogin` file after the remediation:
```
session     required                  pam_lastlog.so showfailed
session     optional                   pam_umask.so silent
session     [success=1 default=ignore] pam_succeed_if.so service !~ gdm* service !~ su* quiet
session     [default=1]                pam_lastlog.so nowtmp showfailed
session     optional                   pam_lastlog.so silent noupdate showfailed
```
It means that failed logins will be shown in any case, but successful login won't be shown three times, as shown in the example:

```
$ ssh root@rhel8
Last login: Wed Aug  9 08:35:44 EDT 2023 from 192.168.122.1 on pts/0
Last failed login: Wed Aug  9 08:39:51 EDT 2023 from 192.168.122.1 on ssh:notty
There were 3 failed login attempts since the last successful login.
Last login: Wed Aug  9 08:39:55 2023 from 192.168.122.1
```

This is still duplicating information without any additional value. After some tests it was concluded that default settings are pretty much reasonable and no changes are necessary.

#### Addition Information/Debugging Steps:
- https://bugzilla.redhat.com/show_bug.cgi?id=2157044
- https://github.com/ComplianceAsCode/content/pull/9009
- https://www.stigviewer.com/stig/red_hat_enterprise_linux_8/2022-12-06/finding/V-230381
- https://linux.die.net/man/8/pam_lastlog

**Comment by marcusburghardt** [comment]
Here is the explanation of the `/etc/pam.d/postlogin` file as it is by out-of-box:
```
session     optional                   pam_umask.so silent
session     [success=1 default=ignore] pam_succeed_if.so service !~ gdm* service !~ su* quiet
session     [default=1]                pam_lastlog.so nowtmp showfailed
session     optional                   pam_lastlog.so silent noupdate showfailed
```

OBS.: All lines are relevant only for the session phase of the authentication.

`session     optional                   pam_umask.so silent`

- `pam_umask.so` - this module sets the file mode creation mask for the session. It adjusts the default permissions for files and directories created during the session
- the `silent` option means the module will operate silently without producing any messages
- it is `optional` which means its success or failure will not affect the overall authentication process

`session     [success=1 default=ignore] pam_succeed_if.so service !~ gdm* service !~ su* quiet`

- `pam_succeed_if.so` - this module succeeds if a specified condition is met
- `service !~ gdm* service !~ su* quiet` - the condition will succeed if the service name is not (!~) matching gdm* and not matching su*
- the `quiet` option means the module will operate quietly without producing any messages
- `[success=1 default=ignore]` - this is a control statement that specifies how to handle the result of the module.
  - `success=1` - if the module succeeds, skip the next line (skip one line)
  - `default=ignore` - if the module fails, ignore the failure and continue processing

`session     [default=1]                pam_lastlog.so nowtmp showfailed`

- `pam_lastlog.so` - this module records user login activity and displays the last login information.
- [default=1]` - this control statement specifies that if the module returns a failure result, move on to the next line (skip one line).
- the `nowtmp` option will display last login time but won't update the /var/log/wtmp file
- the `showfailed` option will display a message for failed login attempts

`session     optional                   pam_lastlog.so silent noupdate showfailed`

- second call of `pam_lastlog.so`, which will only be 
- the `silent` option means the module will operate silently without producing any messages
- the `noupdate` option means the module will display last login time but do not update the /var/log/wtmp file
- the `showfailed` option will display a message for failed login attempts
- it is `optional` which means its success or failure will not affect the overall authentication process

Now, lets interpret the last three lines, which are the most relevant for `pam_lastlog`:

1. if the service name is `not (!~) matching gdm* and not matching su*`, which is usually the case for interactive users, skip the next (success=1) line and go directly to the last line:
  `session     optional                   pam_lastlog.so silent noupdate showfailed`
  the result here is to only show something if there were any previously failed login (showfailed), otherwise it will execute silently (silent) and won't update any file.

2. if the service name matches `gdm* or su*` the condition will fail and line will be ignored (default=ignore), going directly to the next line:
  `session     [default=1]                pam_lastlog.so nowtmp showfailed`
  the result here is to show previous failed and successful logins without updating the `/var/log/wtmp` file. Then, because the ` [default=1] `, the last line will be skipped.

**Comment by marcusburghardt** [comment]
The `pam_lastlog.so` module is only used in `/etc/pam.d/postlogin` file:
```
# grep -ri "pam_lastlog.so" /etc/pam.d/*
/etc/pam.d/postlogin:session     [default=1]                pam_lastlog.so nowtmp showfailed
/etc/pam.d/postlogin:session     optional                   pam_lastlog.so silent noupdate showfailed
```

But the `/etc/pam.d/postlogin` file is included in many other PAM files:
```
# grep -ri "postlogin" /etc/pam.d/*
/etc/pam.d/login:auth       include      postlogin
/etc/pam.d/login:session    include      postlogin
/etc/pam.d/passwd:password   substack	postlogin
/etc/pam.d/remote:auth       include      postlogin
/etc/pam.d/remote:session    include      postlogin
/etc/pam.d/sshd:auth       include      postlogin
/etc/pam.d/sshd:session    include      postlogin
/etc/pam.d/su:auth		include		postlogin
/etc/pam.d/su:session		include		postlogin
```

This makes more clear the reason to have a conditional for `gdm` and `su` in  `/etc/pam.d/postlogin` file.
Unlike programs like `login` and `ssh`, for example, they needs the `pam_lastlog.so` to display last login information after successful logins.

During some tests, even including `silent` in all `pam_lastlog.so` lines from `/etc/pam.d/postlogin` file, the logins via TTY and SSH always show the `Last login: ...` information. In other words, the default configuration of `/etc/pam.d/postlogin` complements the `gdm` and `su` programs to also show the  `Last login: ...` information while not repeating the information for other programs which already show the information regardless of `pam_lastlog.so`.

**Comment by marcusburghardt** [comment]
So, in summary:

The `pam_lastlog.so` module is used in the `/etc/pam.d/postlogin` file (or login file depending on the distro) to display `Last login` information for certain programs. Although this module is only explicitly mentioned in the `postlogin` file, it is indirectly used by various other PAM configuration files, such as those for `login`, `ssh`, `passwd`, and others. The reason for this is to ensure that the `Last login` information is shown for programs like `gdm` and `su`, while avoiding redundancy for programs that already display this information without needing `pam_lastlog.so`. Despite the usage of `silent` option in one line, the default configuration of `/etc/pam.d/postlogin` ensures consistent display of `Last login` information for specific programs.

In short we only need to check if at least one line contains `am_lastlog.so`, `showfailed` and not `silent`.

This PR better shows how the check and remediation should be: #10979
